packages/core/admin/admin/src/hooks/useAdminUsers/useAdminUsers.js

```diff
@@ -0,0 +1,38 @@
+import { useQuery } from 'react-query';
+import { useFetchClient } from '@strapi/helper-plugin';
+import { stringify } from 'qs';
+
+export function useAdminUsers(params = {}, queryOptions = {}) {
+  const { id = '', ...queryParams } = params;
+  const queryString = stringify(queryParams, { encode: false });
+
+  const { get } = useFetchClient();
+
+  const { data, isError, isLoading, refetch } = useQuery(
+    ['users', id, queryParams],
+    async () => {
+      const {
+        data: { data },
+      } = await get(`/admin/users/${id}${queryString ? `?${queryString}` : ''}`);
```

Comment:
this could be a follow up but i think we should start using the `params` config option of a request alongside the `paramsSerializer` option – https://axios-http.com/docs/req_config to avoid having to do what you're doing _all the time_. wdyt?

Reply:
I'd love to but `params` can't handle the URL structure we have (deep nested objects and arrays). Axios follows UrlSearchParams standards, whereas we have invented our own using `qs`.

I have a topic for the next FE sync for that, because the way we concatenate URLs is very dangerous and annoying.

Reply:
<img width="638" alt="CleanShot 2023-05-24 at 14 52 35@2x" src="https://github.com/strapi/strapi/assets/37798644/7bb45797-99c9-46cf-8ecc-ff6ed84c1fc5">

But this function would let us use `qs` no? yeah topic for FE sync ofc

Reply:
Ah - from what I understood in https://github.com/axios/axios/issues/738 we'd have to use our own serializer, which is a great idea, but I'd like to introduce this in `getFetchClient` instead, so that it applies for all of them.

Let's discuss in the FE sync.